services/src/main/java/org/keycloak/theme/ExtendingThemeManager.java

```diff
@@ -172,6 +176,8 @@ private Theme findTheme(String name, Theme.Type type) {
     }
 
     public static class ExtendingTheme implements Theme {
+        private final static Pattern ENVIRONMENT_VARIABLE_SUBSTITUTE_PATTERN = Pattern.compile("\\$\\{env.([A-Za-z0-9_.-]+)(?::([^\\}]*))?\\}");
```

Comment:
This code doesn't belong here. Take a look at:
* org.keycloak.common.util.SystemEnvProperties
* org.keycloak.common.util.StringPropertyReplacer

Reply:
Ok thanks for the advice ! 
With these two classes I got rid of the dirty things I added to ExtendingTheme class 👍